extensions/DNin/wake-lock.js

```diff
@@ -77,9 +87,14 @@
         // Not supported in this browser.
         return;
       }
+      const enable = Scratch.Cast.toBoolean(args.enabled);
+      if (enable && document.visibilityState === "hidden") {
+        // Can't request wake lock while document is hidden.
```

Comment:
what happens if you try anyways?

Reply:
Error:
```
DOMException: Failed to execute 'request' on 'WakeLock': The requesting page is not visible
```

From there, it catches the error and sets `latestEnabled` to false so it can try again. But I kinda feel like we shouldn't have any errors or time wasted trying to acquire a wake lock when we know we can't. What do you think?